src/DynamoForgeDataExchange/DynamoForgeDataExchange.cs

```diff
@@ -0,0 +1,333 @@
+            // RETRIEVE BACK THE UPLOADED BINARY ASSET AND CONVERT TO DYNAMO GRAPH
+            // STEP 1: Download the binary asset from a specified exchange - this call will return an AWS download URL
+            var downloadBinaryUrl = $"/v1/collections/{CollectionId}/exchanges/{ExchangeContainerId}/assets/binaries:download";
+            RestRequest downloadBinaryRequest = new RestRequest(downloadBinaryUrl);
+
+            var downloadBinaryAsset = new UploadedBinaryAsset(BinaryAssetGuid);
+            var downloadBinaries = new BinaryAssets();
+            downloadBinaries.AddBinary(downloadBinaryAsset);
+            string binaryDownloadBody = JsonConvert.SerializeObject(downloadBinaries);
+
+            downloadBinaryRequest.AddStringBody(binaryDownloadBody, ContentType.Json);
+            AddHeadersToPostRequest(downloadBinaryRequest, token);
+            var downloadBinaryResponse = client.ExecutePost(downloadBinaryRequest);
+            dynamic downloadBinaryResponseBody = JObject.Parse(downloadBinaryResponse.Content);
+            var downloadURL = downloadBinaryResponseBody["binaries"][0]["downloadUrl"].Value;
+
+            // STEP 2: Retrieve the binary data using the returned AWS download URL
+            var fileDownloadClient = new RestClient();
+            RestRequest downloadDataRequest = new RestRequest(downloadURL);
+            var downloadDataResponse = fileDownloadClient.ExecuteGet(downloadDataRequest); // Data returned in base 64 compressed
+            var base64Data = downloadDataResponse.Content;
+
+            // STEP 3: Convert base64 compressed data back to .dyn file
+            byte[] compressedData = Convert.FromBase64String(base64Data);
+            byte[] uncompressedData = DataUtilities.Decompress(compressedData);
+            string uncompressedString = Encoding.UTF8.GetString(uncompressedData);
+            string parentDir = Directory.GetParent(filePath).FullName;
+            var dynamoParsedPath = Path.Combine(parentDir, "MaximizeWindowViews-deserialized.dyn");
+            File.WriteAllText(dynamoParsedPath, uncompressedString);
+        }
+
+        internal static void DataExchange(string filePath)
+        {
+            // The Forge team has recommended to use the Stage environment for testing.
+            // Depending on whether we are using Stage or Prod, a Forge token needs to be retrieved 
+            // using client and secret id from a Forge app created in the respective environment.
+            // Assuming there is a way to retrieve the 3-leg Forge token in Dynamo, I will use a hardcoded one here.
+            var token = GetAuthorizationToken();
+            var stageClientUrl = "https://developer-stg.api.autodesk.com/exchange";
+            //var prodClientUrl = "https://developer.api.autodesk.com/exchange";
+
+            // Stage collectionId created for Dynamo
+            CollectionId = "9R09ArUBUEDVRIGQ5OE373_L2C"; // for staging
+            CollectionId = ""; // for production
```

Comment:
Can we use a configuration file for these? Also seems like the `CollectionId ` is being re-assined to empty string, but the Url above is for staging?

Reply:
yeah will do